### ISSUE TITLE
[Workflow] In edit, Enter should cause jump to next field

### ISSUE THREAD

**Issue by twMat** [Jermolene/TiddlyWiki5#1709]
A very "front-end" and "every-day" issue:

For smoother workflow in edit mode, pressing Enter should behave (somewhat) like the TAB key, i.e "focus on next field" - for the fields _title_ and _tags_. 

Note that currently, pressing Enter there does nothing in those fields.

Differences from the TAB key:
TAB is designed to jump between every detail in the field, which for the "tags procedure" makes a very long road if you just want to reach the text field. It even opens the tagpill dropdown and jumps between every tagpill. This means TAB is totally unpractical for quick jump from tag field to text field.

Some specifics:

In the tags field, if focus is on an _empty_ tag field (i.e where the user can begin to type a tag name), then pressing Enter should leave the tag field empty - and cause jump to the text field.

If the user has begun to type in a tagname string, and presses Enter, then this is made into a tag(pill) - and focus is shifted to the text field.

If the user is standing on a tagpill in the dropdown and presses Enter, then this tagpill is selected (i.e it becomes a tag) - and focus is shifted to the text field.

**Comment by tobibeer** [comment]
@twMat , thanks for raising this issue. Keyboard usability is actually a big one for me. Moving the mouse at every step is very ineffective and I would hope TiddlyWiki behaved more like an effective editor via keyboard for one who wishes to edit it, than the current pointer-driven ui.

For proper distinction, this is how I envistion ENTER and TAB to work.

* **TAB**
    * move to the next **input** field
    * not to help links, not to add buttons, not to select-popups
* **ENTER**
    * when there's something to actually be commited, do that but stay where you are...
        * add a new tag
        * add a new field-value pair after having entered the value
    * a newline in
        * the tiddler text area
        * any multiline-fields (#1947, https://github.com/Jermolene/TiddlyWiki5/issues/1932#issuecomment-141637499)

> This means TAB is totally unpractical for quick jump from tag field to text field.

It shouldn't be, and that's why I don't think a force commit via ENTER is the solution. I think the TAB key needs to work optimally, that is to move to the next **main** input.

> In the tags field, if focus is on an empty tag field (i.e where the user can begin to type a tag name), then pressing Enter should leave the tag field empty - and cause jump to the text field.
> If the user has begun to type in a tagname string, and presses Enter, then this is made into a tag(pill) - and focus is shifted to the text field.

I think when the field is empty nothing should happen and when it contains text, a tag should be added, the field emptied, and given new focus ...so that the user can enter another tag.

Same for new fields. After hitting ENTER in the new field-value input, the field should be added as if the add button was pressed and then the new field-name input should be focussed anew.

In other words, ENTER should never move to the next field, TAB needs to do that properly.

> If the user is standing on a tagpill in the dropdown and presses Enter, then this tagpill is selected (i.e it becomes a tag) - and focus is shifted to the text field.

Almost. When in the new-tag input, hitting the DOWN key should focus the select popup where one can select a tag via DOWN, UP and finally ENTER. Once a tag is added, the focus should **not** move to the text field but instead **stay** in the new-tag input ...for adding another tag.

**Comment by pmario** [comment]
If we want to have a proper cursor / enter key handling, we would need to implement our own keyboard handling and completely disable it on the OS level. .. Even windows native programs don't have proper navigation key handling. ... 

---- rant

All going back to the one developer, that thought the tab key could be used for this, while they should have removed this completely useless key, together with the windows key. ..

**Comment by tobibeer** [comment]
> we would need to implement our own keyboard handling and completely disable it on the OS level

true and doable ...although I wouldn't know if and how that itch can be scrateched with widgets alone.

**Comment by BurningTreeC** [comment]
@twMat , is this issue resolved in the current prerelease?

**Comment by twMat** [comment]
@BurningTreeC 

>@twMat , is this issue resolved in the current prerelease?

Yes, issue now superfluous. (And your excellent work makes it a joy to navigate around with the keyboard!)
